File: q20-q29/q24/q24.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def swapPairs(self, head: ListNode) -> ListNode:
        if head == None:
            return None
        if head.next == None:
            return head
        solution = ListNode(0)
        solution.next = head
        first = head
        second = head.next
        first.next = second.next
        second.next = first
        solution.next = second
        cur_node = first
        while first.next != None and first.next.next != None:
            first = first.next
            second = first.next
            first.next = second.next
            second.next = first
            cur_node.next = second
            cur_node = first
        return solution.next

File: Sequence/q1-q99/q20-q29/q24/q24.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
#include <iostream>
class Solution {
public:
    ListNode* swapPairs(ListNode* head) {
        if (head == NULL)
            return NULL;
        if (head->next == NULL)
            return head;
        ListNode* solution = new ListNode(0);
        solution->next = head;
        ListNode* first = head;
        ListNode* second = head->next;
        first->next = second->next;
        second->next = first;
        solution->next = second;
        ListNode* cur_node = first;
        while (first->next != NULL && first->next->next != NULL){
            first = first->next;
            second = first->next;
            first->next = second->next;
            second->next = first;
            cur_node->next = second;
            cur_node = first;
        }
        return solution->next;
    }
};
